apps/crc_proposal_end.py

```diff
@@ -16,36 +16,26 @@
 class CrcProposalEnd(BaseParser):
     """Display the end date for an account's current CRC proposal."""
 
-    banking_db_path = 'sqlite:////ihome/crc/bank/crc_bank.db'
-
     def __init__(self) -> None:
         """Define arguments for the command line interface"""
 
         super(CrcProposalEnd, self).__init__()
-
         default_group = Shell.run_command("id -gn")
-        self.add_argument(
-            'account', default=default_group, nargs='?',
-            help=f'the name of the Slurm account [default: {default_group}]')
+        help_text = f"SLURM account name [defaults to your primary group: {default_group}]"
+        self.add_argument('account', nargs='?', default=default_group, help=help_text)
 
-    def get_proposal_end_date(self, account: str) -> datetime:
+    def get_proposal_end_date(self, account: str) -> str:
         """Get the proposal end date for a given account
 
         Args:
             account: The name of the account
 
         Returns:
-            The proposal end date as a ``datetime`` object
+            The proposal end date in string format
         """
 
-        database = dataset.connect(self.banking_db_path, sqlite_wal_mode=False)
-        table = database['proposal']
-
-        db_record = table.find_one(account=account)
-        if db_record is None:
-            self.error(f"The account: {account} doesn't appear to exist")
-
-        return db_record['end_date']
+        acct = AccountServices(account)
```

Comment:
This is abstraction is significantly better. In the long run we should consider exposing a DAO (or equivalent) in the banking package so we don't rely on private methods. Looks perfect for now.